src/runtime/prrte_globals.c

```diff
@@ -808,7 +808,7 @@ static void prrte_proc_construct(prrte_proc_t* proc)
     proc->app_rank = -1;
     proc->last_errmgr_state = PRRTE_PROC_STATE_UNDEF;
     proc->state = PRRTE_PROC_STATE_UNDEF;
-    proc->app_idx = 0;
+    proc->app_idx = UINT32_MAX;
```

Comment:
I believe we default to 0 for the specific reasons that (a) multi-app jobs are rare, and (b) we don't need an "undefined" value. Are you testing for this somewhere?

Reply:
I'm testing locally.

What I'm seeing happen is that "procs" (the daemon?) are being inserted into the node->procs array at index 0, and then prun gets confused checking when comparing app->idx to proc->app_idx. So by defaulting this to non-zero, those stray procs won't have the same index (0) since their proc->app_idx is never set.

Reply:
Yeah, but that means you are band-aiding over a different error with another one. The correct answer is that you should be checking the job, not just the app_idx of the proc, to ensure you are looking at the correct proc_t. Those "stray" procs are tools that have attached from the node - they will have a very different jobid.

Reply:
I can try - but I am not sure that will work with multi-app jobs. I think for jobs with multiple apps, like I was testing, we need to compare the app idx as well.